pkg/provider/helm/values.go

```diff
@@ -4,171 +4,430 @@
+		csids["default"] = map[string]any{
+			"enabled": false,
+		}
+
+		// Adds the attestation policies as ClusterSPIFFEID CRs to be reconciled by the spire-controller-manager.
 		for _, binding := range g.trustZone.AttestationPolicies {
 			policy, err := g.source.GetAttestationPolicy(binding.Policy)
 			if err != nil {
 				return nil, err
 			}
+
 			clusterSPIFFEIDs, err := attestationpolicy.NewAttestationPolicy(policy).GetHelmConfig(g.source, binding)
 			if err != nil {
 				return nil, err
 			}
-			spireServerValues[fmt.Sprintf(`"spire-server"."controllerManager"."identities"."clusterSPIFFEIDs"."%s"`, policy.Name)] = clusterSPIFFEIDs
+
+			csids[policy.Name] = clusterSPIFFEIDs
 		}
-	} else {
-		// defaults to true
-		spireServerValues[`"spire-server"."controllerManager"."identities"."clusterSPIFFEIDs"."default"."enabled"`] = true
 	}
 
-	// add federations as clusterFederatedTrustDomains to be reconcilced by spire-controller-manager
+	// Adds the federations as ClusterFederatedTrustDomain CRs to be reconciled by the spire-controller-manager.
 	if len(g.trustZone.Federations) > 0 {
 		for _, fed := range g.trustZone.Federations {
 			tz, err := g.source.GetTrustZone(fed.To)
 			if err != nil {
 				return nil, err
 			}
+
 			if tz.GetBundleEndpointUrl() != "" {
-				spireServerValues[`"spire-server"."federation"."enabled"`] = true
-				config := federation.NewFederation(tz).GetHelmConfig()
-				spireServerValues[fmt.Sprintf(`"spire-server"."controllerManager"."identities"."clusterFederatedTrustDomains"."%s"`, fed.To)] = config
+				spireServer["federation"] = map[string]any{
+					"enabled": true,
+				}
+
+				cftd, ok := getNestedMap(identities, "clusterFederatedTrustDomains")
+				if !ok {
+					identities["clusterFederatedTrustDomains"] = map[string]any{
+						fed.To: federation.NewFederation(tz).GetHelmConfig(),
+					}
+				} else {
+					cftd[fed.To] = federation.NewFederation(tz).GetHelmConfig()
+				}
 			}
 		}
 	}
 
-	spiffeOIDCDiscoveryProviderValues := map[string]interface{}{
-		`"spiffe-oidc-discovery-provider"."enabled"`: false,
+	soidcpv := spiffeOIDCDiscoveryProviderValues{enabled: false}
+	spiffeOIDCDiscoveryProviderValues, err := soidcpv.generateValues()
+	if err != nil {
+		return nil, err
 	}
 
-	spiffeCSIDriverValues := map[string]interface{}{
-		`"spiffe-csi-driver"."fullnameOverride"`: "spiffe-csi-driver",
+	scsidv := spiffeCSIDriverValues{fullnameOverride: "spiffe-csi-driver"}
+	spiffeCSIDriverValues, err := scsidv.generateValues()
+	if err != nil {
+		return nil, err
 	}
 
-	valuesMaps := []map[string]interface{}{
+	valuesMaps := []map[string]any{
 		globalValues,
 		spireAgentValues,
 		spireServerValues,
 		spiffeOIDCDiscoveryProviderValues,
 		spiffeCSIDriverValues,
 	}
 
-	ctx := cuecontext.New()
-	combinedValuesCUE := ctx.CompileBytes([]byte{})
+	combinedValues := flattenMaps(valuesMaps)
+
+	if g.values != nil {
+		combinedValues = mergeMaps(g.values, combinedValues, true)
```

Comment:
nit: Comparing with the standard `append()` function, the arguments seem backwards